### PR TITLE
Fixes #22156 - Repo discovery table showing empty rows

### PR DESCRIPTION
**Pull request by sjha4**
### Description of problem:

Sometimes the Repo discovery page has table with empty rows. 

****
### Steps to Reproduce: 
1. Navigate to Content > Products
2. Click on Repo Discovery.
3. Chances are that you will see a table of empty rows (Number of empty rows being equal to number of products)

##### If unable to reproduce with above steps:
1. Go to Content > Product
2. Click on any product in the table.
3. Click on Product link on Breadcrumb
4. Before the Product table loads, click on Repo Discovery.

### CODE REVIEW

**Comment by sjha4** [comment]
[test katello]

**Comment by jturel** [comment]
@sjha4 this seems to work. I am noticing some issues in JS console while using chrome that were present before this change. I'm wondering if you have plans to fix and/or if there are open issues to track it:
```
bastion-0804cdbab99e30c73521ca9e24c759dbbe365e08ea9de03312106991c88ef13d.js:30285 TypeError: Cannot read property '0' of null
```
```
[DOM] Found 2 elements with non-unique id #currentPage: (More info: https://goo.gl/9p2vKq) 
```

**Comment by sjha4** [comment]
@jturel: The console errors/warnings are because the same table is being used for products and the repo discovery page. I have created an [issue](http://projects.theforeman.org/issues/23381) around the same to track. I believe some changes might be needed in Bastion for that but I am not entirely sure.

File: engines/bastion_katello/app/assets/javascripts/bastion_katello/products/products.controller.js [hunk 1]
```diff
@@ -47,10 +47,16 @@ angular.module('Bastion.products').controller('ProductsController',
...
-        nutupane = new Nutupane(Product, params);
+        nutupaneParams = {
+            'disableAutoLoad': true
+        };
+        $scope.disableRepoDiscovery = true;
+        nutupane = new Nutupane(Product, params, undefined, nutupaneParams);
         $scope.controllerName = 'katello_products';
         nutupane.masterOnly = true;
-
+        nutupane.refresh().then(function () {
```
Comment (jturel) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/products/products.controller.js]: Sorry I didn't catch this the first time I looked, but I think this is a behavior worth having a test for - can you add one here?

**Comment by jturel** [comment]
Gotcha. Thanks for opening an issue!

File: engines/bastion_katello/app/assets/javascripts/bastion_katello/products/products.controller.js (hunk 1 above)
Comment (sjha4) [reply engines/bastion_katello/app/assets/javascripts/bastion_katello/products/products.controller.js]: @jturel: Added some tests. Also, some linting that Rubymine did for me.

**Comment by jlsherrill** [comment]
Thanks @sjha4

**Pull request merged**